sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -691,7 +691,7 @@ object DataSource extends Logging {
           val internalSources = sources.filter(_.getClass.getName.startsWith("org.apache.spark"))
           if (provider.equalsIgnoreCase("xml") && sources.size == 2) {
             val externalSource = sources.filterNot(_.getClass.getName
-              .startsWith("org.apache.spark.sql.execution.datasources.xml.XmlFileFormat")
+              .startsWith("org.apache.spark.sql.execution.datasources.v2.xml.XmlDataSourceV2")
```

Comment:
Is the discovered source `XmlDataSourceV2` irrespective of whether xml is included in the USE_V1_SOURCE_LIST?

Reply:
Yes, `ServiceLoader` uses `META-INF.services/org.apache.spark.sql.sources.DataSourceRegister`  file to load data source. Whether it's v1 or v2 it's the same here. 

I can get your  doubts. When the `USE_V1_SOURCE_LIST ` includes `XML`, we finally back to using the default FileFormat here.

https://github.com/apache/spark/blob/9e35d04cb030ce2ac736914a5586d159339b4686/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala#L111